src/wallet/test/coinselector_tests.cpp

```diff
@@ -298,8 +298,11 @@ BOOST_AUTO_TEST_CASE(bnb_search_test)
         /*tx_noinputs_size=*/ 0,
         /*avoid_partial=*/ false,
     };
-    coin_selection_params_bnb.m_change_fee = coin_selection_params_bnb.m_effective_feerate.GetFee(coin_selection_params_bnb.change_output_size);
-    coin_selection_params_bnb.m_cost_of_change = coin_selection_params_bnb.m_effective_feerate.GetFee(coin_selection_params_bnb.change_spend_size) + coin_selection_params_bnb.m_change_fee;
+
+    // Cost of change is the cost of creating the change output + cost of spending the change output in the future.
+    // cost of change = (change output size * effective feerate) + (size of spending change output * discard feerate)
```

Comment:
I do not agree with replacing the variables, but If you believe that as it is the code is not easily understandable, why not just add those comments?

Reply:
Why not both?  Just adding the comments would help though if you're really that interested in having the redundant computations run every time the test suit is run.

Reply:
> Then why not also change the next line to:
```
 // 204 = (68 * 3000) / 1000)
coin_selection_params_bnb.min_viable_change = CAmount(204)
```

I think that's a good idea.  It was a while ago when I put in this PR but I think I didn't look into that parameter at the time.

Reply:
@yancyribbens I have a slightly different perception about whether the computations are redundant. As I see it, the advantage of unit tests is that we ensure that obvious cases have the obvious results that we expect them to have, every time.

Reply:
@davidgumberg that's fair.  I've updated this PR based on what @kouloumos said.  I'm not sure if you agree that this is easier to reason about or not which was the motivation for creating this PR.